src/client.ts

```diff
@@ -859,34 +866,50 @@ export class ZkBobClient {
     }
     for (let index = 0; index < txParts.length; index++) {
       const onePart = txParts[index];
-      if (onePart.outNotes.length != 1) {
+      
+      let oneTxData: any;
+      let txType: TxType;
+      if (onePart.outNotes.length == 0) {
+        const oneTx: ITransferData = {
+          outputs: [],
+          fee: onePart.fee.toString(),
+        };
+        oneTxData = await state.createTransferOptimistic(oneTx, optimisticState);
+        txType = TxType.Transfer;
+      } else if (onePart.outNotes.length == 1) {
+        const oneTx: IWithdrawData = {
+          amount: onePart.outNotes[0].amountGwei.toString(),
+          fee: onePart.fee.toString(),
+          to: addressBin,
+          native_amount: '0',
+          energy_amount: '0',
+        };
+        oneTxData = await state.createWithdrawalOptimistic(oneTx, optimisticState);
+        txType = TxType.Withdraw;
+      } else {
         throw new Error('Invalid transaction configuration');
       }
-      const onePartAmount = onePart.outNotes[0].amountGwei;
-      const oneTx: IWithdrawData = {
-        amount: onePartAmount.toString(),
-        fee: onePart.fee.toString(),
-        to: addressBin,
-        native_amount: '0',
-        energy_amount: '0',
-      };
-      const oneTxData = await state.createWithdrawalOptimistic(oneTx, optimisticState);
 
       const startProofDate = Date.now();
       const txProof: Proof = await this.proveTx(tokenAddress, oneTxData.public, oneTxData.secret);
       const proofTime = (Date.now() - startProofDate) / 1000;
       console.log(`Proof calculation took ${proofTime.toFixed(1)} sec`);
 
-      const transaction = {memo: oneTxData.memo, proof: txProof, txType: TxType.Withdraw};
+      const transaction = {memo: oneTxData.memo, proof: txProof, txType};
 
       const jobId = await this.sendTransactions(token.relayerUrl, [transaction]);
       this.startJobMonitoring(tokenAddress, jobId);
       jobsIds.push(jobId);
 
       // Temporary save transaction part in the history module (to prevent history delays)
       const ts = Math.floor(Date.now() / 1000);
-      const record = await HistoryRecord.withdraw(address, onePartAmount, onePart.fee, ts, '0', true);
-      state.history.keepQueuedTransactions([record], jobId);
+      if (txType == TxType.Transfer) {
+        const record = await HistoryRecord.transferOut([], onePart.fee, ts, '0', true, (addr) => this.isMyAddress(tokenAddress, addr));
```

Comment:
UI most likely won't display this correctly, we'll need to update it as well

last argument seems unused
```suggestion
        const record = await HistoryRecord.transferOut([], onePart.fee, ts, '0', true, () => false);
```

Reply:
Yes, we definitely need to adjust ui. Fixed: https://github.com/zkBob/zkbob-client-js/pull/89/commits/2f6b8fae70b30f12ca2a0b7711eca949702ecbe7